libtsuba/test/rdg-slice.cpp

```diff
@@ -0,0 +1,124 @@
+#include "katana/Result.h"
+#include "tsuba/RDGManifest.h"
+#include "tsuba/RDGSlice.h"
+
+namespace {
+// This test tests the following:
+// 1. loading/unloading properties and checking the schema work (don't crash)
+// 2. loading/unloading properties changes the properties tables as expected
+// 3. loading/unloading non-existent properties behaves as expected
+katana::Result<void>
+TestPropertyLoading(const std::string& path_to_manifest) {
+  tsuba::RDGManifest manifest =
+      KATANA_CHECKED(tsuba::FindManifest(path_to_manifest));
+  tsuba::RDGHandle rdg_handle =
+      KATANA_CHECKED(tsuba::Open(std::move(manifest), tsuba::kReadOnly));
+  // RDGFile will close the handle on destroy
+  tsuba::RDGFile handle(rdg_handle);
+
+  // this arg doesn't load any useful topology, but we are only testing property
+  // loading and unloading, so this should be fine
+  tsuba::RDGSlice::SliceArg slice_arg{
+      .node_range = std::make_pair(0, 1),
+      .edge_range = std::make_pair(0, 1),
+      .topo_off = 0,
+      .topo_size = 0};
+
+  std::vector<std::string> no_props;
+  auto rdg_slice = KATANA_CHECKED(
+      tsuba::RDGSlice::Make(rdg_handle, slice_arg, 0, no_props, no_props));
+
+  // input is ldbc 003
+  int64_t expected_num_node_props = 17;
+  int64_t expected_num_edge_props = 3;
+  KATANA_LOG_ASSERT(
+      rdg_slice.full_node_schema()->num_fields() == expected_num_node_props);
+  KATANA_LOG_ASSERT(rdg_slice.node_properties()->num_columns() == 0);
+  KATANA_LOG_ASSERT(
+      rdg_slice.full_edge_schema()->num_fields() == expected_num_edge_props);
+  KATANA_LOG_ASSERT(rdg_slice.edge_properties()->num_columns() == 0);
+
+  // load all properties
+  for (int64_t i = 0; i < expected_num_node_props; ++i) {
+    KATANA_CHECKED(rdg_slice.load_node_property(
+        rdg_slice.full_node_schema()->field(i)->name()));
```

Comment:
While this code is natural to write, it causes the repeated creation of the full schema.  I wonder if it would be pedagogically superior to `auto full_schema = rdg_slice.full_node_schema();` and then use `full_schema `in the loop.

Reply:
I would be open to that. But I kind of like this style. One of the things that this version tests is that loading properties doesn't affect the full schema. If we changed to your proposal, we would have to explicitly test that somewhere else. I suppose explicit tests are probably better than implicit tests.

Reply:
Well, at least you thought about it.  If you are checking that loading properties doesn't change the full schema, it probably merits a comment.